server/events/models/models.go

```diff
@@ -369,16 +374,96 @@ type PlanSuccess struct {
 type PolicySetResult struct {
 	PolicySetName  string
 	ConftestOutput string
+	PolicyHash     string
 	Passed         bool
 	ReqApprovals   int
 	CurApprovals   int
 }
 
-// PolicySetApproval tracks the number of approvals a given policy set has.
-type PolicySetStatus struct {
-	PolicySetName string
-	Passed        bool
-	Approvals     int
+// Encapsulate PolicySetResult in a custom type
+type PolicySetDataList []PolicySetResult
+
+// Compress compresses ConftestOutput on the parent for storage in the backend.
+func (p PolicySetDataList) GetCompressed() PolicySetDataList {
+	var policySetDataList PolicySetDataList
+	for i, policySetResult := range p {
+		var conftestOutput string
+		if policySetResult.ConftestOutput != "" {
+			var b bytes.Buffer
+
+			// Errors are highly unlikely and DB update logic is a bit hairy, just panic if errors occur.
+			gz := gzip.NewWriter(&b)
+			_, err := gz.Write([]byte(policySetResult.ConftestOutput))
+			if err != nil {
+				panic(err)
```

Comment:
I'm worried how Atlantis handles the panics in the call stack. Can we make sure we have a test case setup for this?

Reply:
> Just a request for additional testing for panicing on Gzip data to/from the backend. @runatlantis/maintainers thoughts on this when handle policies? Do we panic for anything else in Atlantis or try to wrap to avoid service disruptions?

@GenPage - Originally, I had tried to pass an error around and handle it in the backend (boltdb, redis), but the current state of logic would place handling these errors after operations had occurred and from what I could tell the backend data representing a PR might end up in an incongruent state. It could be possible to re-work the backend logic to account for this. But, I was also having a hard time seeing what would cause gzip or base64 to error out in a closed loop as is implemented here. Definitely open to thoughts, suggestions.